### DISCUSSION TITLE
Question: what is the usual interface to program to the create3, which is intended from the manufacturer?

### DISCUSSION THREAD

**Discussion by jbhFriends** [iRobotEducation/create3_docs#504] [Questions]
### How are you connecting to your Create 3?

via wifi to a controller mounted on top.    (initially used the create3 wifi to connect it to the controller, which is intended to be a permanent connection)

### Computer(s) Model(s) and Operating System(s)

dell - windows

### Which version of ROS 2 is installed on your computer?

None

### Which firmware version is installed on your robot?

H.2.3

### Which RMW is your robot running?

FastRTPS

### Is the robot connected to a network? If so, what is the network type?

it connects to a controller for robotics mounted on top. we wifi into the controller and issue commands there.

### Are there multiple Create 3 robots connected to your network?

No

### Is multicast enabled?

No

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

So we have just recently gotten a Create3 from a robotics org, with their controller which connects to it. We have not yet successfully made the create3 move via code through the controller. We seem to have a connection between the two entities.  Since we purchased our gear from the robotics group, we have no connection to the manufacturer.

our question is, from manufacturer side, what is the method / IDE, etc used to control the create3 robot? and it's instructions. 
does the existence of our additional controller make it impossible to use the robot as intended by manufacturer? 

essentially we would like to expand our knowledge of this create3 by using it stand-alone as well as in conjunction with our controller. is this possible?

**Comment by shamlian** [comment]
Hi @jbhFriends ,

I see no reason why your Create 3 should not work as intended once unplugged and off network from the third party packaged controller. You may need to factory reset it or update its firmware in the case that the repackaged unit's settings have been modified in an unexpected way.

As far as the "method or IDE used to control the Create 3 robot," please refer to the iRobot Education [Learning Library](https://edu.irobot.com/learning-library?robotValue=Advanced%20Create%203%20Robot&toggle=lessons). The two primary means of command and control are using ROS 2 over a network or the iRobot BLE protocol with iRobot Coding, the Python Web Playground, or the Python SDK.

**Comment by jbhFriends** [comment]
Hi!
thank you for responding. I think the only changes I made to it were: updating to h.2.3
-Change the "Address and port of Fast DDS discovery server to:
192.168.125.1:11811
-Under the ROS 2 Parameters File, change safety_override from "none" to "full".
-Also add: reflexes_enabled: false
-rmw_fast-cpp  on the dropdown.

My questions:
-What are these things?
-Are they  reset-required-to-use-it-your-way  level of changes?

Using windows, what is the no-hassle approach to getting python working with it/it’s sensors.
And anything else you might like to add.

We very much appreciate your time to answer our questions!!
jack

**Comment by shamlian** [comment]
Please read the documentation and lessons I linked; it took a long time to compose them, and I think they will answer your questions.

tl;dr for you is to use https://python.irobot.com/ to control your robot over Bluetooth Low Energy using a Chrome-based browser.
